typescript/api/services/I18nEntriesService.ts

```diff
@@ -343,12 +365,13 @@ export module Services {
     }
 
   public async syncEntriesFromBundle(bundleOrId: any, overwrite = false): Promise<void> {
-      const bundle = _.isString(bundleOrId)
-        ? await I18nBundle.findOne({ id: bundleOrId })
-        : bundleOrId;
+      const bundle = (bundleOrId && typeof bundleOrId === 'object' && 'data' in bundleOrId && 'locale' in bundleOrId)
```

Comment:
I suggest creating a Typescript narrowing function for Bundle instead of checking inline like this.
Something like this: https://github.com/redbox-mint/redbox-portal/blob/feature/ng19-forms_record-view/packages/sails-ng-common/src/config/form-types.outline.ts#L18